test/end-to-end/offer.js

```diff
@@ -286,6 +291,41 @@ describe("Offer endpoint tests", () => {
                 await Offer.deleteMany({});
             });
 
+            beforeEach(async () => {
+                await Offer.deleteMany({});
+            });
+
+            test("Should fail to create an offer due to publish end date being six months after publish date", async () => {
```

Comment:
The title should be generic so that if we change the limit, we don't have to change the test. Instead of "six months", change to "being after publish date more than the limit"